tawazi/_dag/dag.py

```diff
@@ -750,13 +644,41 @@ def __post_init__(self) -> None:
                     "cache_deps_of can't be used together with target_nodes or exclude_nodes"
                 )
 
-            self.graph = self.dag.make_subgraph(target_nodes=self.cache_deps_of)
+            cache_deps_of = (
+                self.dag.get_multiple_nodes_aliases(self.cache_deps_of)
+                if self.cache_deps_of is not None
+                else self.cache_deps_of
+            )
+            self.graph = self.dag.graph_ids.make_subgraph(target_nodes=cache_deps_of)
         else:
-            self.graph = self.dag.make_subgraph(
-                target_nodes=self.target_nodes,
-                exclude_nodes=self.exclude_nodes,
-                root_nodes=self.root_nodes,
+            # clean user input
+            target_nodes = (
+                self.dag.get_multiple_nodes_aliases(self.target_nodes)
+                if self.target_nodes is not None
+                else self.target_nodes
+            )
+            exclude_nodes = (
+                self.dag.get_multiple_nodes_aliases(self.exclude_nodes)
+                if self.exclude_nodes is not None
+                else self.exclude_nodes
             )
+            root_nodes = (
+                self.dag.get_multiple_nodes_aliases(self.root_nodes)
+                if self.root_nodes is not None
+                else self.root_nodes
+            )
+
+            self.graph = self.dag.graph_ids.make_subgraph(
+                target_nodes=target_nodes, exclude_nodes=exclude_nodes, root_nodes=root_nodes
+            )
+
+        if cfg.RUN_DEBUG_NODES:
+            debug_ids = self.dag.graph_ids.include_debug_nodes(self.graph.leaf_nodes)
+            self.graph = self.dag.graph_ids.subgraph(list(self.graph.nodes) + debug_ids).copy()
+
+        # 3. Remove debug nodes
+        else:
+            self.graph.remove_nodes_from([node_id for node_id in self.graph.debug_nodes])
```

Comment:
there is some repetition between what we have here and what we have in DAG.setup. I suggest to separate DAG.make_subgraph into two parts (1st include target_nodes, exclude_nodes, root_nodes and the 2nd to add debug nodes) . This way we can reuse the code and not duplicate it

Reply:
same comment as the one before concerning debug nodes

Reply:
is there no way to refactor this ? maybe using a function ?

Reply:
well we could but it would save like 6 lines, I think our best bet would maybe be to leave it as is and rewrite it when we refactor the DExec since we'll likely get rid of the duplication at this point

Reply:
@bashirmindee same here